packages/datetime2/src/components/date-input2/dateInput2.tsx

```diff
@@ -568,13 +568,7 @@ export const DateInput2: React.FC<DateInput2Props> = React.memo(function _DateIn
 
     // We use the renderTarget API to flatten the rendered DOM and make it easier to implement features like the "fill" prop.
     const renderTarget = React.useCallback(
-        // N.B. pull out `defaultValue` so that it's not forwarded to the DOM.
```

Comment:
note that this is another benefit of the type changes in https://github.com/palantir/blueprint/pull/5802 -- we no longer have to deal with extraneous props which are not actually passed to the target renderer, but were presumed to exist in the previous type definitions 👍🏽